backend/backend.go

```diff
@@ -43,7 +43,7 @@ type Backend interface {
 
 	// CreateOrchestrationInstance creates a new orchestration instance with a history event that
 	// wraps a ExecutionStarted event.
-	CreateOrchestrationInstance(context.Context, *HistoryEvent) error
+	CreateOrchestrationInstance(context.Context, *HistoryEvent, *protos.CreateInstanceOption) error
```

Comment:
This require interface updates, is there a better way to do this. 
I read about `context.WithValue` to pass data, but seems it's not very recommended.